qiskit/chemistry/algorithms/pes_samplers/potentials/potential_base.py

```diff
@@ -0,0 +1,175 @@
+        Default units (scaling=1.0) are Angstroms. Scale by 1E-10 to get
+        meters.
+
+        Args:
+            scaling: scaling factor
+
+        Returns:
+            equilibrium geometry
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_minimal_energy(self, scaling: float = 1.0) -> float:
+        """Get the minimal energy.
+
+        Returns the value of the minimal energy (scaled by 'scaling')
+        Default units (scaling=1.0) are J/mol. Scale appropriately for
+        Hartrees.
+
+        Args:
+            scaling: scaling factor
+
+        Returns:
+            minimum energy
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_trust_region(self) -> Tuple[float, float]:
+        """Get the trust region.
+
+        Returns the bounds of the region (in space) where the energy
+        surface implementation can be trusted. When doing spline
+        interpolation, for example, that would be the region where data
+        is interpolated (vs. extrapolated) from the arguments of
+        fit().
+
+        Returns:
+            the trust region between bounds
+        """
+        raise NotImplementedError
+
+
+class VibronicStructureBase(ABC):
+    """
+    Class to hold a molecular vibronic structure providing access to
+    vibrational modes and energy levels.
+    """
+
+    def __init__(self, molecule: Molecule) -> None:
+        self.update_molecule(molecule)
+
+    def update_molecule(self, molecule: Molecule) -> Molecule:
+        """
+        Wipe state if molecule changes, and check validity of molecule
+        for potential.
+        Args:
+            molecule: chemistry molecule
+
+        Returns:
+            molecule used
+        """
+        self.molecule = molecule
+
+    @abstractmethod
+    def get_num_modes(self) -> float:
+        """Returns the number of vibrational modes for the molecule.
+
+        Returns:
+            the number of vibrational modes
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def vibrational_energy_level(self, n: int, mode: int = 0) -> float:
+        """Returns the n-th vibrational energy level for a given mode.
+
+        Args:
+            n: number of vibrational mode
+            mode: vibrational mode
+
+        Returns:
+            n-th vibrational energy level for a given mode
+        """
+        raise NotImplementedError
+
+    def get_maximum_trusted_level(self, mode: int = 0) -> float:  # pylint: disable=unused-argument
+        """
+        Returns the maximum energy level for which the particular
+        implementation still provides a good approximation of reality.
+        Default value of 100. Redefined where needed (see e.g. Morse).
+
+        Args:
+            mode: vibronic mode
+
+        Returns:
+            maximum_trusted_level setted
+        """
+        return 100
+
+
+class PotentialBase(EnergySurfaceBase, VibronicStructureBase):
```

Comment:
Parent VibronicStructureBase has an init with a molecule. This has no way to pass that up (related to comments in the MoresPotential)

Reply:
should be fixed with the invocation of the `super().update_molecule()`.